### PR TITLE
docs: change the theme

### PR DESCRIPTION
**Pull request by 12rambau**
@mariobuikhuizen, I would like to work on the documentation and update the template that is used. my objective are the following : 
- [x] use the pydata-sphinx-theme (because thats my favorite for sphinx) 
- refactor the structure of the documentation to expose the API 
- try to pull information from the vuetify API
- use the widgetti logo

This PR is just a demo using pydata-sphinx-theme, is there a chance to see this PR merged if I continue working on it ? 

Related to #268

### CODE REVIEW

**Comment by maartenbreddels** [comment]
I like the theme and your plans!

**Comment by 12rambau** [comment]
requires #240 to move forward

**Comment by 12rambau** [comment]
@mariobuikhuizen I'll stop this PR here to avoid a too big scope. what is cover is the following: 
- adapt deps to use the pydata-sphinx-theme
- adapt conf.py accordingly (and remove automatically generated comments that are not there anymore since Sphinx 6) 
- refactor the toctree to make it conform to pydata-sphinx-theme

work that I will do in the following-up PR: reorder pages to make better use of the theme sidebars

**Comment by mariobuikhuizen** [comment]
Can you please remove the changes not related to changing the theme?

**Comment by 12rambau** [comment]
I scanned the modifications again and I see 2 potential candidates: 
- drop of requirements.txt (not used anywhere since #240)
- drop the extensive comments present in the conf.py when I refactored it to include the pydata-sphinx-theme config. 

Which should I remove ?

**Comment by mariobuikhuizen** [comment]
To be clear, those changes are good to do, but not in the PR called "change the theme".

**Comment by 12rambau** [comment]
then the answer is "both of them" 😄  Understood, I'll create other PR for them

**Comment by mariobuikhuizen** [comment]
Thank you 😄

**Comment by 12rambau** [comment]
let's wait for the test completion but normally I'm good to go

**Comment by mariobuikhuizen** [comment]
One of the advantages of keeping changes as concise as possible is less chance of conflicts. #272 also changes code formatting: https://github.com/widgetti/ipyvuetify/commit/32273a0c0c0d7f24ff5e8e7444f55a2ceeff3dcb#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R25 , which leads to a conflict here.

Also, I see now that that PR removes `master_doc = "index"`, this is certainly something different than removing comments.

**Comment by 12rambau** [comment]
conflict resolved. `master-doc` is an ignored parameter if there is an "index.rst" next to conf.py. I didn't even realized I removed it.

**Comment by mariobuikhuizen** [comment]
If I check the build on: https://ipyvuetify--238.org.readthedocs.build/en/238/ I don't see the new theme applied

File: docs/conf.py [hunk 1]
```diff
@@ -23,3 +23,35 @@
 html_theme = "sphinx_rtd_theme"
```
Comment (12rambau) [inline docs/conf.py:23]:
```suggestion
html_theme = "pydata_sphinx_theme"
```

**Review by mariobuikhuizen — APPROVED** [review]
Much cleaner diff now! 😄

**Comment by mariobuikhuizen** [comment]
Thanks!

**Comment by 12rambau** [comment]
ok I'll try to be as fast as possible to make the follow-up PR to have a more readable documentation ASAP

**Comment by mariobuikhuizen** [comment]
What less readable now?

**Comment by mariobuikhuizen** [comment]
Come to think of it I can also point the stable branch to just before the theme change commit. That will give more time to make your changes.

**Comment by mariobuikhuizen** [comment]
Production docs are back to the old theme.

**Pull request merged**